signer/cosigner_security_ecies_test.go

```diff
@@ -58,3 +75,58 @@ func testCosignerSecurity(t *testing.T, securities []CosignerSecurity) error {
 
 	return err
 }
+
+func BenchmarkCosignerECIES(b *testing.B) {
+	keys := make([]*ecies.PrivateKey, 3)
+	pubs := make([]*ecies.PublicKey, 3)
+
+	for i := 0; i < 3; i++ {
+		key, err := ecies.GenerateKey(rand.Reader, secp256k1.S256(), nil)
+		require.NoError(b, err)
+
+		keys[i] = key
+		pubs[i] = &key.PublicKey
+	}
+
+	securities := make([]CosignerSecurity, 3)
+
+	for i := 0; i < 3; i++ {
+		securities[i] = NewCosignerSecurityECIES(CosignerECIESKey{
+			ID:        i + 1,
+			ECIESKey:  keys[i],
+			ECIESPubs: pubs,
+		})
+	}
+
+	for i := 0; i < b.N; i++ {
+		var eg errgroup.Group
+		for i, security := range securities {
+			security := security
+			i := i
+			eg.Go(func() error {
+				var nestedEg errgroup.Group
+				for j, security2 := range securities {
+					if i == j {
+						continue
+					}
+					security2 := security2
+					j := j
+					nestedEg.Go(func() error {
+						n, err := security.EncryptAndSign(j+1, []byte("mock_pub"), []byte("mock_share"))
+						if err != nil {
+							return err
+						}
+
+						_, _, err = security2.DecryptAndVerify(i+1, n.PubKey, n.Share, n.Signature)
+						if err != nil {
+							return err
+						}
+						return nil
+					})
+				}
+				return nestedEg.Wait()
+			})
+		}
+		require.NoErrorf(b, eg.Wait(), "success count: %d", i)
```

Comment:
It's unusual to see an errgroup (especially 2) within a benchmark test. I'm sure @mark-rushakoff may have some thoughts here. Why not run everything serially? 

Additionally, splitting benchmark tests between the methods `EncryptAndSign` and `DecryptAndVerify` will help isolate which methods become slower or faster over time.

Subtests may be useful here. https://go.dev/blog/subtests (Or may not 😄 )

Reply:
The errgroups were used here to simulate horcrux' runtime usage of the `CosignerSecurity`. It helped to reproduce the issue in the ecies/go/v2 lib. A benchmark may not be the best tool for this, but many thousands of iterations were required in some cases to reproduce the issue.

Reply:
Great idea, I updated them to be tests instead of benchmarks with a reasonable number of iterations.